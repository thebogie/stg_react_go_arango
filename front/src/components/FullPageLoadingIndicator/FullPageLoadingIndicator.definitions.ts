export interface FullPageLoadingIndicatorProps {
  title?: string;
}
