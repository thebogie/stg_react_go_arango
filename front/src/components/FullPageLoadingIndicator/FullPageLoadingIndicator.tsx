import PropagateLoader from "react-spinners/PropagateLoader";
import type { FullPageLoadingIndicatorProps } from "./FullPageLoadingIndicator.definitions";
import "./FullPageLoadingIndicator.styles.css";

/**
 * @component FullPageLoadingIndicator
 * @description A full page loading indicator that displays a progress bar and a title.
 * @param {string} title - The title to display below the progress bar (optional - defaults to "Loading")
 */
const FullPageLoadingIndicator = ({ title }: FullPageLoadingIndicatorProps) => (
  <div
    data-testid="fullpageloadingindicator-testId"
    className="full-page-loading-indicator"
  >
    <div className="full-page-loading-indicator-progress">
      <PropagateLoader color={"#76e7aa"} />
    </div>
    {title && <div className="full-page-loading-indicator-title">{title}</div>}
  </div>
);

export default FullPageLoadingIndicator;
