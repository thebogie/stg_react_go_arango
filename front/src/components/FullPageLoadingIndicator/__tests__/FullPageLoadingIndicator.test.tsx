import FullPageLoadingIndicator from "../FullPageLoadingIndicator";
import { render } from "@testing-library/react";

describe("FullPageLoadingIndicator Tests", () => {
  test("correctly renders snapshot", () => {
    const component = render(<FullPageLoadingIndicator />);
    expect(component).toMatchSnapshot();
  });
  test("correctly renders the full page loading indicator", () => {
    // Render the full page loading indicator
    const component = render(<FullPageLoadingIndicator />);
    // Obtain the full page loading indicator by its testId
    const fullPageLoadingIndicator = component.getByTestId(
      "fullpageloadingindicator-testId"
    );
    expect(fullPageLoadingIndicator).toBeTruthy();
  });
  test("correctly renders the full page loading indicator with a custom title", () => {
    // Render the full page loading indicator
    const component = render(
      <FullPageLoadingIndicator title={"Custom Title"} />
    );
    // Obtain the full page loading indicator by its testId
    const fullPageLoadingIndicator = component.getByTestId(
      "fullpageloadingindicator-testId"
    );
    // Check that the title is rendered correctly
    expect(fullPageLoadingIndicator.children[1].innerHTML).toBe("Custom Title");
  });
});
