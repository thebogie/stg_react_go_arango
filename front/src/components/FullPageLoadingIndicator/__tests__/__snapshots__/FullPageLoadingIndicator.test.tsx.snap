// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullPageLoadingIndicator Tests correctly renders snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="full-page-loading-indicator"
        data-testid="fullpageloadingindicator-testId"
      >
        <div
          class="full-page-loading-indicator-progress"
        >
          <span
            style="display: inherit; position: relative;"
          >
            <span
              style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(118, 231, 170); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-0 1.5s infinite; animation-fill-mode: forwards;"
            />
            <span
              style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(118, 231, 170); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-1 1.5s infinite; animation-fill-mode: forwards;"
            />
            <span
              style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(118, 231, 170); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-2 1.5s infinite; animation-fill-mode: forwards;"
            />
            <span
              style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(118, 231, 170); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-3 1.5s infinite; animation-fill-mode: forwards;"
            />
            <span
              style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(118, 231, 170); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-4 1.5s infinite; animation-fill-mode: forwards;"
            />
            <span
              style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(118, 231, 170); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-5 1.5s infinite; animation-fill-mode: forwards;"
            />
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="full-page-loading-indicator"
      data-testid="fullpageloadingindicator-testId"
    >
      <div
        class="full-page-loading-indicator-progress"
      >
        <span
          style="display: inherit; position: relative;"
        >
          <span
            style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(118, 231, 170); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-0 1.5s infinite; animation-fill-mode: forwards;"
          />
          <span
            style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(118, 231, 170); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-1 1.5s infinite; animation-fill-mode: forwards;"
          />
          <span
            style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(118, 231, 170); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-2 1.5s infinite; animation-fill-mode: forwards;"
          />
          <span
            style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(118, 231, 170); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-3 1.5s infinite; animation-fill-mode: forwards;"
          />
          <span
            style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(118, 231, 170); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-4 1.5s infinite; animation-fill-mode: forwards;"
          />
          <span
            style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(118, 231, 170); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-5 1.5s infinite; animation-fill-mode: forwards;"
          />
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
